pkg/steps/loggingclient/loggingclient.go

```diff
@@ -0,0 +1,121 @@
+package loggingclient
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type LoggingClient interface {
+	ctrlruntimeclient.Client
+	// Object contains the latest revision of each object the client has
+	// seen. Calling this will reset the cache.
+	Objects() []ctrlruntimeclient.Object
+}
+
+func New(upstream ctrlruntimeclient.Client) LoggingClient {
+	return &loggingClient{
+		upstream: upstream,
+		logTo:    map[string]map[string]ctrlruntimeclient.Object{},
+	}
+}
+
+type loggingClient struct {
+	lock     sync.Mutex
+	upstream ctrlruntimeclient.Client
+	logTo    map[string]map[string]ctrlruntimeclient.Object
+}
+
+func (lc *loggingClient) Get(ctx context.Context, key ctrlruntimeclient.ObjectKey, obj ctrlruntimeclient.Object) error {
+	if err := lc.upstream.Get(ctx, key, obj); err != nil {
+		return err
+	}
+	lc.logObject(obj)
+	return nil
+}
+
+func (lc *loggingClient) List(ctx context.Context, list ctrlruntimeclient.ObjectList, opts ...ctrlruntimeclient.ListOption) error {
+	// TODO: Should we support the logging here? We'd need to use reflect to get the Items field. Its reasonable safe as controller-runtime
+	// itself will bail out if that is not present
+	return lc.upstream.List(ctx, list, opts...)
+}
+
+func (lc *loggingClient) Create(ctx context.Context, obj ctrlruntimeclient.Object, opts ...ctrlruntimeclient.CreateOption) error {
+	if err := lc.upstream.Create(ctx, obj, opts...); err != nil {
+		return err
+	}
+	lc.logObject(obj)
+	return nil
+}
+
+func (lc *loggingClient) Delete(ctx context.Context, obj ctrlruntimeclient.Object, opts ...ctrlruntimeclient.DeleteOption) error {
+	if err := lc.upstream.Delete(ctx, obj, opts...); err != nil {
+		return err
+	}
+	lc.logObject(obj)
```

Comment:
Does this log the object that we had in memory which we used to issue the Delete call or does Delete respond with something? The glaring issue with our current artifacts is that when we Delete we get no object, and this code will at least give us "some" version of that object, but it would be amazing to get the *latest* version of it so we can debug with status, etc.

Reply:
err good point, that does indeed not work. We could just get it first here, wdyt?

Reply:
Updated the client to get -> log -> delete inside its delete